### PR TITLE
fix(plugin-storybook): remove dist dir when build

### PR DESCRIPTION
**Pull request by targeral**
## Description

remove dist dir when run `modern build`

## Types of changes

- [ ] Docs change / Dependency upgrade
- [x] Bug fix
- [ ] New feature / Improvement
- [ ] Refactoring
- [ ] Breaking change

## Checklist

- [x] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/cli/plugin-storybook/src/index.ts [hunk 1]
```diff
@@ -39,6 +40,19 @@ export default (pluginOption: PluginOptions = {}): CliPlugin<ModuleTools> => ({
       return config;
     },
 
+    beforeBuild: async () => {
+      const { fs } = await import('@modern-js/utils');
+      const { STORYBOOK_DIST_DIR_NAME } = await import('./features/constants');
+      const { appDirectory } = api.useAppContext();
+      const storybookDistPath = path.join(
+        appDirectory,
+        'dist',
```
Comment (chenjiahan) [inline packages/cli/plugin-storybook/src/index.ts:49]: We can use`appContext.distDirectory`

Comment (targeral) [reply packages/cli/plugin-storybook/src/index.ts:49]: How is `distDirectory` calculated? It may not have much to do with Module Tools

Comment (targeral) [reply packages/cli/plugin-storybook/src/index.ts:49]: This output path is currently temporary and an api should be designed later.

Comment (chenjiahan) [reply packages/cli/plugin-storybook/src/index.ts:49]: You are right, ignore this comment.

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**